Source/RxCBCentralManager.swift

```diff
@@ -109,22 +109,9 @@ class RxCBCentralManager: RxCentralManagerType {
 	}
 
 	/// Current central manager state
-	var state: CBCentralManagerState {
-		return centralManager.state
-	}
-
-	/// Current continuous state of Central Manager
-	var rx_state: Observable<CBCentralManagerState> {
-		return centralManager
-			.rx_observeWeakly(CBCentralManagerState.self, "state")
-			.flatMap {
-                state -> Observable<CBCentralManagerState> in
-				guard let state = state else {
-					return Observable.error(BluetoothError.BluetoothInUnknownState)
-				}
-				return Observable.just(state)
-            }
-            .replay(1)
+	var state: BluetoothState {
+        guard let bleState = BluetoothState(rawValue: centralManager.state.rawValue) else { return .Unsupported }
```

Comment:
`return BluetoothState(rawValue: centralManager.state.rawValue) ?? .Unsupported`

would be more swifty I guess..